vulnerable.cpp

```
@@ -1 +1,108 @@
+        t2.join();
+        cout << "Global Counter: " << globalCounter << endl;
+
+        // 透過 new 配置記憶體後，delete 後仍使用（野指標問題）
+        int *dangling = new int(42);
+        delete dangling;
+        cout << "Dangling pointer value: " << *dangling << endl;
+
+        // 分配記憶體後，重複釋放（雙重釋放）
+        int *doubleFreePtr = new int(100);
+        delete doubleFreePtr;
+        delete doubleFreePtr;
```

Cppcheck (2.10-2)

[error] Memory pointed to by 'doubleFreePtr' is freed twice.

(doubleFree)